### ISSUE TITLE
A used function should have had coverage mapping data but did not

### ISSUE THREAD

**Issue by Zalathar** [rust-lang/rust#118850]
The immediate symptoms of #118643 were fixed by #118666, but some users reported that their builds now encounter another coverage-related ICE:

https://github.com/rust-lang/rust/issues/118643#issuecomment-1849690436
https://github.com/rust-lang/rust/issues/118643#issuecomment-1849962689

```
error: internal compiler error: compiler/rustc_codegen_llvm/src/coverageinfo/mapgen.rs:98:17: A used function should have had coverage mapping data but did not: _RNvMsl_NtNtNtCsl0gwtrD5SWs_5winit13platform_impl8platform15window_delegateNtB5_19WinitWindowDelegate21window_did_resign_key
```

**Comment by Zalathar** [comment]
This ICE does not occur on current stable/beta, but does occur on nightly.

@rustbot label +regression-from-stable-to-nightly

**Comment by Zalathar** [comment]
(This will be fixed by #118852, which is awaiting review.)

**Comment by apiraino** [comment]
WG-prioritization assigning priority ([Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/.23118850.20A.20used.20function.20should.20have.20had.20coverage.20mapping.20da.E2.80.A6)).

@rustbot label -I-prioritize +P-high
